### ISSUE TITLE
Remove log4j1 usage from hazlcast

### ISSUE THREAD

**Issue by tharindu1st** [hazelcast/hazelcast#20330]
[

**Please describe the problem you are trying to solve**
In current hazlcast dependencies, it packs the log4j1 related dependencies inside in jar.
Since log4j1 retired in 2015 as per [1], Is there any plan to release a new hazlcast version, removing the log4j1 related dependencies?.

**Please describe the desired behavior**

[1] - https://logging.apache.org/log4j/1.2/

**Describe alternatives you've considered**
...
](https://issues.apache.org/jira/browse/LOGGING-181

**Comment by mmedenjak** [comment]
Hi @tharindu1st ! Which hazelcast version are you using?

**Comment by tharindu1st** [comment]
Hi @mmedenjak ,
We are using 3.12.2

**Comment by jerrinot** [comment]
hello @tharindu1st,

thank you for trying to make Hazelcast better and taking time to fill this issue! 

Let me explain the situation with Hazelcast logging: Hazelcast [integrates](https://docs.hazelcast.com/hazelcast/5.0/maintain-cluster/logging#hide-nav) with a number of logging libraries & logging facades, including log4j2, slf4j and log4j1. Hazelcast will use log4j1 for logging if and only if explicitly instructed to do so via configuration and log4j1 must be available on a classpath. 

Hazelcast does NOT distribute artifacts containing log4j1 JAR(s) and Hazelcast does not have a hard dependency on log4j1. Hazelcast Platform distributions (=container image, zip, etc) contain log4j2. The JAR distribution does NOT have a hard-dependency on any specific logging library/facade. It defaults to JUL and users can explicitly configure Hazelcast to use a different library if they choose to do so. 

In other words: Hazelcast can integrate with log4j1, but it's always a deliberate choice of a user. The main purpose is to allow integration with bigger projects which are staying on log4j1 for any reason. 

I'm closing this issue, feel free to re-open it if you have further question!
